bokehjs/src/lib/models/plots/plot_canvas.ts

```diff
@@ -867,9 +867,13 @@ export class PlotView extends LayoutDOMView {
       this.connect(rng.change, () => {this._needs_layout = true; this.request_paint()})
     }
 
-    this.connect(this.model.properties.renderers.change, async () => {
-      await this.build_renderer_views()
-    })
+    this.connect(this.model.properties.above.change, async () => {await this.build_renderer_views()})
+    this.connect(this.model.properties.below.change, async () => {await this.build_renderer_views()})
+    this.connect(this.model.properties.left.change, async () => {await this.build_renderer_views()})
+    this.connect(this.model.properties.right.change, async () => {await this.build_renderer_views()})
+    this.connect(this.model.properties.center.change, async () => {await this.build_renderer_views()})
+    this.connect(this.model.properties.renderers.change, async () => {await this.build_renderer_views()})
```

Comment:
This can be simplified to:
```ts
const {above, below, left, right, center, renderers} = this.model.properties
this.on_change([above, below, left, right, center, renderers], async () => await this.build_renderer.views())
```

Reply:
I don't really like the repetition - both here and in the `build_renderer_views` function. Well, I tried to be clever and here's what I came up with:
```typescript
  _get_main_renderers_attrs<T extends Plot | Plot.Props>(x: T): T extends Plot ? Renderer[][] : p.Property<Renderer[]>[]  {
    return <T extends Plot ? Renderer[][] : p.Property<Renderer[]>[]> [x.above, x.below, x.left, x.right, x.center, x.renderers]
  }
```
A bit too much, isn't it? Also, no clue why the `<...>` appears to be required after `return`.

Reply:
>  don't really like the repetition (...)

This kind of repetition isn't great, but my concern is primarily with duplicated implementations, which may give readers wrong impressions about this code.

> A bit too much, isn't it?

A bit, yet tempting, Also I think it may be a bit confusing the user of such method. Depending on the context, there are other useful combinations of those properties:

- above, below, left, right - side panel renderers
- *, center - panel renderers
- *, renderers - all renderers

so we may need more than one method. Still may be useful if there are many call sites that would use them.

> Also, no clue why the <...> appears to be required after return.

This is because whether using a type parameter or not, `x` is of type `Plot | Plot.Props` to this implementation and thus `x.property_name` is a union. Then you have to use `T` to disambiguate.

Reply:
I see, thanks! I'll just go with your initial proposition then.